env0/resource_environment.go

```diff
@@ -550,20 +550,66 @@ func createVariable(configurationVariable *client.ConfigurationVariable) any {
 }
 
 // Validate that the template is assigned to the "project_id".
-func validateTemplateProjectAssignment(d *schema.ResourceData, apiClient client.ApiClientInterface) error {
+func validateTemplateProjectAssignment(d *schema.ResourceData, template *client.Template) error {
```

Comment:
need the template in one of the upper functions, so here instead of getting it we pass it here.
we get it from the upper functions.